program/rust/src/time_machine_types.rs

```diff
@@ -7,10 +39,91 @@ pub struct TimeMachineWrapper {
     place_holder: [u8; 1864],
 }
 
+impl Tracker for TimeMachineWrapper {
+    fn add_first_price(
+        &mut self,
+        _current_time: u64,
+        _current_price: i64,
+        _current_conf: u64,
+    ) -> Result<(), OracleError> {
+        msg!("implement me");
+        Ok(())
+    }
+    fn add_price(
+        &mut self,
+        _current_time: u64,
+        _prev_time: u64,
+        _current_price: i64,
+        _prev_price: i64,
+        _current_conf: u64,
+        _prev_conf: u64,
+    ) -> Result<(), OracleError> {
+        msg!("implement me");
+        Ok(())
+    }
+}
+
+#[derive(Copy, Clone)]
```

Comment:
you probably don't want `Copy` right? we likely want every piece of code to point to the instance of this struct that's in the solana account? if you derive copy, you create the room for bugs because the struct got copied under the hood.

Reply:
Copy is required by bytmuck